src/lib.rs

```diff
@@ -321,31 +321,31 @@ fn has_chroma(bo: &BlockOffset, bsize: BlockSize,
 // predict, transform, quantize, write coefficients to a bitstream,
 // dequantize, inverse-transform.
 pub fn encode_tx_block(fi: &FrameInvariants, fs: &mut FrameState, cw: &mut ContextWriter,
-                  p: usize, bo: &BlockOffset, mode: PredictionMode, tx_type: TxType,
+                  p: usize, bo: &BlockOffset, mode: PredictionMode, tx_size: TxSize, tx_type: TxType,
                   po: &PlaneOffset, skip: bool) {
     let stride = fs.input.planes[p].cfg.stride;
     let rec = &mut fs.rec.planes[p];
-    let tx_size = TxSize::TX_4X4;
 
     mode.predict(&mut rec.mut_slice(po), tx_size);
 
     if skip { return; }
 
-    let mut residual = [0 as i16; 16];
+    let mut residual = [0 as i16; 64*64];
```

Comment:
Any way to get the size from TxSize? Otherwise, we need to change this value again when we add bigger tx later.
Same to below, coeffs_storage[], roceffs[].
I believe easiest way is just to allocate max tx size supported though.

Reply:
It's stack allocated so it needs to be constant size. But I could use constant values rather than magic numbers here.